src/components/SingleArticlePage.jsx

```diff
@@ -0,0 +1,35 @@
+import { useParams } from "react-router"
+import { useState, useEffect } from "react"
+import { Link } from "react-router-dom"
+import { fetchArticleById } from "./api"
+
+function SingleArticlePage() {
+    const {article_id} = useParams()
+    const [article, setArticle] = useState([])
+
+    useEffect(() => {
+        fetchArticleById(article_id).then((article) => {
+            setArticle(article.data.article)
```

Comment:
Again, we have two things called article here - only one of them is truly just an article. The other should be renamed.